### PR TITLE
Add function stubs for extension by IHV

### PR DESCRIPTION
**Pull request by tmolenh**

### CODE REVIEW

File: Diagnostics/Get-NetView.PS1 [hunk 1]
```diff
@@ -1712,17 +1712,18 @@ function Worker {
...
     CheckAdminPrivileges $SkipAdminCheck
-    $workDir   = (GetWorkDir $OutputDirectory)
-
-    Clear-Host
+    $workDir   = (NormalizeWorkDir $OutputDirectory)
 
     EnvSetup          -OutDir $workDir
 
+    # Start Run
+    Clear-Host
```
Comment (tmolenh) [inline Diagnostics/Get-NetView.PS1]: Realized the `EnvSetup` can throw an exception based on user input. So the terminal should be cleared after `EnvSetup` so user can see input and exception together.

File: Diagnostics/Get-NetView.PS1 [hunk 2]
```diff
@@ -776,6 +776,48 @@ function NativeNicDetail {
...
+            #"PCI\VEN_1425*" { ChelsioDetail $nic.ifIndex $dir }
+            #"PCI\VEN_15B3*" { MellanoxDetail $nic.ifIndex $dir }
+            #"PCI\VEN_8086*" { IntelDetail $nic.ifIndex $dir }
+            default { <# Not implemented, not native, or N/A #> }
+        }
+    }
+} # NicVendor()
+
+# function stub for extension by IHV
+function MyVendorDetail {
```
Comment (omarcardona) [inline Diagnostics/Get-NetView.PS1]: Move this function definition to before "NicVendor()"

File: Diagnostics/Get-NetView.PS1 [hunk 3]
```diff
@@ -776,6 +776,48 @@ function NativeNicDetail {
     }
 } # NativeNicDetail()
 
+function NicVendor {
```
Comment (omarcardona) [inline Diagnostics/Get-NetView.PS1]: NIT: rename to NicIhvDetail() to follow existing naming convention.

File: Diagnostics/Get-NetView.PS1 [hunk 4]
```diff
@@ -776,6 +776,48 @@ function NativeNicDetail {
...
+    )
+
+    $dir = Join-Path $OutDir "NicVendor"
+    New-Item -Path $dir -ItemType Directory | Out-Null
+
+    foreach ($nic in Get-NetAdapter) {
+        $pciId = (Get-NetAdapterAdvancedProperty -Name $nic.Name -AllProperties -RegistryKeyword "ComponentID").RegistryValue
+        switch -Wildcard($pciId) {
+            # Not implemented
+            #"PCI\VEN_1425*" { ChelsioDetail $nic.ifIndex $dir }
```
Comment (omarcardona) [inline Diagnostics/Get-NetView.PS1]: NIT: Reference NicVendor() callout in at least one of these.

File: Diagnostics/Get-NetView.PS1 [hunk 5]
```diff
@@ -776,6 +776,48 @@ function NativeNicDetail {
...
+            #"PCI\VEN_15B3*" { MellanoxDetail $nic.ifIndex $dir }
```
Comment (omarcardona) [inline Diagnostics/Get-NetView.PS1]: NIT: Avoid single line basic blocks.  Use standard basic blocks as the code elsewhere.

File: Diagnostics/Get-NetView.PS1 (hunk 3 above)
Comment (tmolenh) [inline Diagnostics/Get-NetView.PS1]: Current naming conventions are inconsistent. Went with the concise option.

File: Diagnostics/Get-NetView.PS1 (hunk 4 above)
Comment (tmolenh) [inline Diagnostics/Get-NetView.PS1]: Not sure what you mean here.

File: Diagnostics/Get-NetView.PS1 (hunk 5 above)
Comment (tmolenh) [inline Diagnostics/Get-NetView.PS1]: Fixed

File: Diagnostics/Get-NetView.PS1 (hunk 2 above)
Comment (tmolenh) [inline Diagnostics/Get-NetView.PS1]: Also fixed

File: Diagnostics/Get-NetView.PS1 (hunk 3 above)
Comment (omarcardona) [inline Diagnostics/Get-NetView.PS1]: You're right.  The Detail suffix is inconsistent.

File: Diagnostics/Get-NetView.PS1 (hunk 4 above)
Comment (omarcardona) [inline Diagnostics/Get-NetView.PS1]: Meaning execute the call to a stub function for better clarity.  It's a NIT like I mentioned.

**Pull request merged**